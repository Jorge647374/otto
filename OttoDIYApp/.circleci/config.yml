version: 2
jobs:
  build:
    working_directory: ~/repo
    docker:
      - image: circleci/node:8
    steps:
      - checkout
      - run: yarn
      - persist_to_workspace:
          root: .
          paths:
            - node_modules

  jest_test:
    working_directory: ~/repo
    docker:
      - image: circleci/node:8
    steps:
      - checkout
      - attach_workspace:
          at: ~/repo
      - run: cp .env.test .env
      - run: npm run lint
      - run: npm test
      - store_test_results:
          path: ~/repo/junit.xml

  detox_test:
    # Note: This flow has serious room for time improvment via caching
    macos:
      xcode: "9.0"
    working_directory: ~/repo
    steps:
      - checkout
      - attach_workspace:
          at: ~/repo
      - run: cp .env.test .env

      # Get everything we need to run detox
      - run: brew tap wix/brew
      - run: brew install applesimutils
      - run: npm install -g react-native-cli
      - run: npm install -g detox-cli

      # fix: Error: `fsevents` unavailable
      # https://github.com/expo/expo/issues/854
      - run: npm r -g watchman
      - run: brew install watchman
      - run: npm run test:test-e2e

workflows:
  version: 2
  build-test-deploy:
    jobs:
      - build
      - jest_test:
          requires:
            - build

# temporarily disable e2e tests while the brew formula for python in watchman fixes itself
#      - detox_test:
#         requires:
#            - build
#            - jest_test
#          filters:
#           branches:
#             only:
#               - develop
#               - master
