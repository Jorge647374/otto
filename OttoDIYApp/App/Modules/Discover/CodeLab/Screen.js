import React, { Component } from 'react'
import PropTypes from 'prop-types'

import {
  Container,
  InstructionList,
  CodeLabNav,
  Modal } from 'App/Components'

import { Metrics } from 'App/Themes'

import Types from 'App/Services/PropTypes'

export default class Screen extends Component {
  static propTypes = {
    config: PropTypes.any,
    instructions: PropTypes.arrayOf(PropTypes.shape({
      ...Types.instruction
    })).isRequired,
    showNotConnectedModal: PropTypes.bool.isRequired,
    navHeight: PropTypes.number.isRequired,
    onConnect: PropTypes.func.isRequired,
    onSlidingComplete: PropTypes.func.isRequired,
    onChangeOrder: PropTypes.func.isRequired,
    onNavHeightChange: PropTypes.func.isRequired,
    onClose: PropTypes.func.isRequired,
    onNavPress: PropTypes.func.isRequired,
    onRun: PropTypes.func.isRequired,
    onHideNotConnectedModal: PropTypes.func.isRequired
  }

  resolveSkills = () => {
    const { config } = this.props
    const { moves, skills } = config || {}

    const allSkills = []
    if (moves && moves.length > 0) {
      allSkills.push({
        id: 'moves',
        category: 'Moves',
        items: moves
      })
    }
    if (skills && skills.length > 0) {
      allSkills.push(...skills)
    }

    return allSkills
  }

  render () {
    const {
      instructions,
      showNotConnectedModal,
      navHeight,
      onConnect,
      onSlidingComplete,
      onChangeOrder,
      onNavHeightChange,
      onClose,
      onNavPress,
      onHideNotConnectedModal} = this.props

    const skills = this.resolveSkills()

    return (
      <Container>
        <InstructionList
          style={{marginBottom: navHeight + Metrics.unit}}
          instructions={instructions}
          onSlidingComplete={onSlidingComplete}
          onChangeOrder={onChangeOrder}
          onClose={onClose} />
        <Modal
          navigation={this.props.navigation}
          show={showNotConnectedModal}
          onHidePress={onHideNotConnectedModal}
          onBack={onConnect}
          template='NotConnected' />
        <CodeLabNav
          skills={skills}
          onPress={onNavPress}
          onNavHeightChange={onNavHeightChange} />
      </Container>
    )
  }
}
