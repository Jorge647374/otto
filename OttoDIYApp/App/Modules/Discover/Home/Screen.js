import React, { Component } from 'react'
import { ScrollView } from 'react-native'
import PropTypes from 'prop-types'

import {
  Container,
  Footer,
  Modal,
  Card,
  List,
  FeaturedListItem,
  CompactListItem } from 'App/Components'

import { Metrics, Colors, Images } from 'App/Themes'

export default class Screen extends Component {
  static propTypes = {
    showNotReadyModal: PropTypes.bool.isRequired,
    onHideNotReadyModal: PropTypes.func.isRequired,
    onPress: PropTypes.func.isRequired
  }

  render () {
    const {
      showNotReadyModal,
      onHideNotReadyModal,
      onPress } = this.props
    return (
      <Container>
        <ScrollView style={{ marginBottom: 210 }}>
          <List title='Featured' cols={2}>
            <FeaturedListItem
              image={Images.featured.led}
              title='Blink'
              onPress={() => { onPress('Blink') }} />
            <FeaturedListItem
              icon={'heartbeat'}
              iconColor={Colors.red}
              title='Heart Beat'
              onPress={() => { onPress('Heart Beat') }} />
          </List>
          <List title='Projects'>
            <CompactListItem
              icon='circle-o'
              title='Hello World'
              onPress={() => { onPress('Hello World') }} />
            <CompactListItem
              icon='circle-o'
              title='Dance Show'
              onPress={() => { onPress('Dance Show') }} />
          </List>
        </ScrollView>
        <Modal
          navigation={this.props.navigation}
          show={showNotReadyModal}
          onHidePress={onHideNotReadyModal}
          template='NotReady' />
        <Footer style={{paddingTop: 0}}>
          <Card
            theme='light'
            title='DISCOVER'
            text={'Get inspired by what you can create. Build your own creations or remix others to jumpstart your ideas.'}
            button='Code Lab'
            textAlign='left'
            onPress={() => { onPress('Code Lab') }}
            style={{textView: {marginBottom: Metrics.unit}}} />
        </Footer>
      </Container>
    )
  }
}
