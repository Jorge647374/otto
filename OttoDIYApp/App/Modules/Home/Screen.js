import React, { Component } from 'react'
import PropTypes from 'prop-types'

import { Container, List, ListItem, BottomNav } from 'App/Components'

export default class Screen extends Component {
  static propTypes = {
    onNavigatePress: PropTypes.func.isRequired
  }

  render () {
    const { onNavigatePress } = this.props
    return (
      <Container>
        <List>
          <ListItem
            title='Play & Explore'
            text='Let’s play'
            button='Play'
            onPress={() => { onNavigatePress('PlayerScreen') }} />
          <ListItem
            title='Discover'
            text='Let’s see what you can do'
            button='Discover'
            onPress={() => { onNavigatePress('DiscoverScreen') }} />
          <ListItem
            title='Learn'
            text='You’ll be a guru in no time'
            button='Learn'
            onPress={() => { onNavigatePress('LearnScreen') }} />
          <ListItem
            title='Lab'
            text='Get under the hood'
            button='Lab'
            onPress={() => { onNavigatePress('LabScreen') }} />
        </List>
        <BottomNav
          onRatePress={() => { onNavigatePress('RateScreen') }}
          onHomePress={() => {}}
          onSettingsPress={() => { onNavigatePress('SettingsScreen') }} />
      </Container>
    )
  }
}
