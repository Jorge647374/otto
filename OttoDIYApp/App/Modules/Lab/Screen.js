import React, { Component } from 'react'
import PropTypes from 'prop-types'

import {
  Container,
  Footer,
  Modal,
  Links,
  Link,
  List,
  StatListItem } from 'App/Components'

import { Metrics } from 'App/Themes'

export default class Screen extends Component {
  static propTypes = {
    showNotReadyModal: PropTypes.bool.isRequired,
    onHideNotReadyModal: PropTypes.func.isRequired,
    onPress: PropTypes.func.isRequired
  }

  render () {
    const {
      showNotReadyModal,
      onHideNotReadyModal,
      onPress } = this.props
    return (
      <Container>
        <List style={{ margin: Metrics.unit * 2 }} cols={2}>
          <StatListItem icon='bluetooth' title='bluetooth' value='Connected' />
          <StatListItem icon='battery' title='battery' value='85%' />
          <StatListItem icon='play' title='playtime' value='4.8h' />
          <StatListItem icon='code' title='remixes' value='5' />
          <StatListItem icon='book' title='lessons' value='2' />
          <StatListItem icon='gamepad' title='level' value='1' />
        </List>
        <Modal
          navigation={this.props.navigation}
          show={showNotReadyModal}
          onHidePress={onHideNotReadyModal}
          template='NotReady' />
        <Footer>
          <Links>
            <Link theme='light' text='Shop' onPress={onPress} />
            <Link theme='light' text='Customize' onPress={onPress} />
            <Link theme='light' text='Add a buddy' onPress={onPress} />
            <Link theme='light' text='View logs' onPress={onPress} />
          </Links>
        </Footer>
      </Container>
    )
  }
}
