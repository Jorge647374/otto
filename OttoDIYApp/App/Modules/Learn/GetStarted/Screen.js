import React, { Component } from 'react'
import PropTypes from 'prop-types'

import {
  Container,
  Carousel,
  Card } from 'App/Components'

import { Images } from 'App/Themes'

export default class Screen extends Component {
  static propTypes = {
    connected: PropTypes.bool.isRequired,
    onDone: PropTypes.func.isRequired,
    onConnectPress: PropTypes.func.isRequired
  }

  render () {
    const { connected, onDone, onConnectPress } = this.props

    const connectButton = (!connected)
      ? {title: 'Connect', onPress: onConnectPress}
      : {title: 'Next', onPress: null}

    const connectCard = (!connected)
      ? {title: 'Get connected', text: 'Letâ€™s get you connected & ready to start playing'}
      : {title: 'Nice work, you`re connected', text: null}

    return (
      <Container>
        <Carousel
          buttons={['Next', connectButton.title, 'Done']}
          onPress={[null, connectButton.onPress, null]}
          onDone={onDone}>
          <Card image={Images.hello} title='Hi there ðŸ‘‹' />
          <Card
            image={Images.bluetooth}
            title={connectCard.title}
            text={connectCard.text} />
          <Card
            image={Images.done}
            title='Woot woot ðŸŽ‰'
            text='Youâ€™re all setup & ready to play' />
        </Carousel>
      </Container>
    )
  }
}
