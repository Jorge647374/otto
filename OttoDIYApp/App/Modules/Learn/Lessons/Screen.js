import React, { Component } from 'react'
import { ScrollView } from 'react-native'
import PropTypes from 'prop-types'

import {
  Container,
  Footer,
  Modal,
  Card,
  List,
  ListHeader,
  CompactListItem } from 'App/Components'

import { Metrics } from 'App/Themes'

export default class Screen extends Component {
  static propTypes = {
    showNotReadyModal: PropTypes.bool.isRequired,
    onHideNotReadyModal: PropTypes.func.isRequired,
    onPress: PropTypes.func.isRequired
  }

  render () {
    const {
      showNotReadyModal,
      onHideNotReadyModal,
      onPress } = this.props
    return (
      <Container>
        <ScrollView style={{ marginBottom: 210 }}>
          <List>
            <ListHeader title='Lessons' />
            <CompactListItem
              icon='circle-o'
              title='Get Started'
              onPress={() => { onPress('Get Started') }} />
            <CompactListItem
              icon='circle-o'
              title='Play Time'
              onPress={() => { onPress('Play Time') }} />
            <CompactListItem
              icon='circle-o'
              title='Customize'
              onPress={() => { onPress('Customize') }} />
            <CompactListItem
              icon='circle-o'
              title='Code Remix'
              onPress={() => { onPress('Code Remix') }} />
          </List>
        </ScrollView>
        <Modal
          navigation={this.props.navigation}
          show={showNotReadyModal}
          onHidePress={onHideNotReadyModal}
          template='NotReady' />
        <Footer style={{paddingTop: 0}}>
          <Card
            theme='light'
            title='START YOUR NEXT LESSON'
            text={'Learn them skills!'}
            button='Start'
            textAlign='left'
            onPress={() => { onPress('Get Started') }}
            style={{textView: {marginBottom: Metrics.unit}}} />
        </Footer>
      </Container>
    )
  }
}
