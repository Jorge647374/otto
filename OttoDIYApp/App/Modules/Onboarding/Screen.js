import React, { Component } from 'react'
import PropTypes from 'prop-types'

import {
  Container,
  Carousel,
  Card
} from 'App/Components'

import { Images } from 'App/Themes'

export default class Screen extends Component {
  static propTypes = {
    onDone: PropTypes.func.isRequired
  }

  render () {
    const { onDone } = this.props

    return (
      <Container>
        <Carousel
          buttons={['Next', 'Next', 'Done']}
          onPress={[null, null, null]}
          onDone={onDone}>
          <Card
            image={Images.onboarding.one}
            title='Welcome to the Otto DIY App'
            text='Where you can learn coding and bring your robot to live'
            style={{titleViewText: {fontSize: 22}}}
          />
          <Card
            image={Images.onboarding.two}
            title='Otto awaits your commands'
            text='Otto DIY+, Eyes, Wheels & Humanoid supported out of the box!'
            style={{titleViewText: {fontSize: 22}}}
          />
          <Card
            image={Images.onboarding.three}
            title='Lots more to come! ðŸŽ‰'
            text='Games, more robots, customize your Otto and more fun!'
            style={{titleViewText: {fontSize: 22}}}
          />
        </Carousel>
      </Container>
    )
  }
}
