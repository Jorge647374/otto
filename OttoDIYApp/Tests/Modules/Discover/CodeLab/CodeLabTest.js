import React from 'react'

import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import Screen from 'App/Modules/Discover/CodeLab/Screen'

import config from 'App/Services/Client/Otto/Config'

describe('CodeLab', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallow(
      <Screen
        navigation={{navigate: () => {}}}
        config={config}
        instructions={[]}
        showNotConnectedModal={false}
        navHeight={86}
        onConnect={() => {}}
        onSlidingComplete={() => {}}
        onChangeOrder={() => {}}
        onNavHeightChange={() => {}}
        onClose={() => {}}
        onNavPress={() => {}}
        onRun={() => {}}
        onHideNotConnectedModal={(touch) => {}} />)
  })

  it('Component renders correctly', () => {
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
