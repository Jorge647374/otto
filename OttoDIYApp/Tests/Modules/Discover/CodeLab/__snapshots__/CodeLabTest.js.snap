// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CodeLab Component renders correctly 1`] = `
<Container>
  <InstructionList
    instructions={Array []}
    onChangeOrder={[Function]}
    onClose={[Function]}
    onSlidingComplete={[Function]}
    style={
      Object {
        "marginBottom": 94,
      }
    }
  />
  <Modal
    navigation={
      Object {
        "navigate": [Function],
      }
    }
    onBack={[Function]}
    onHidePress={[Function]}
    show={false}
    template="NotConnected"
  />
  <CodeLabNav
    onNavHeightChange={[Function]}
    onPress={[Function]}
    skills={
      Array [
        Object {
          "category": "Moves",
          "id": "moves",
          "items": Array [
            Object {
              "cmd": "M 0",
              "showDuration": true,
              "title": "Wait",
            },
            Object {
              "cmd": "M 1",
              "showDuration": true,
              "title": "Forwards",
            },
            Object {
              "cmd": "M 2",
              "showDuration": true,
              "title": "Back",
            },
            Object {
              "cmd": "M 4",
              "showDuration": true,
              "title": "Left",
            },
            Object {
              "cmd": "M 3",
              "showDuration": true,
              "title": "Right",
            },
          ],
        },
        Object {
          "category": "Skills",
          "id": "skills",
          "image": Object {
            "process": [Function],
          },
          "items": Array [
            Object {
              "cmd": "M 19",
              "image": Object {
                "process": [Function],
              },
              "stopAtEnd": true,
              "title": "Jitter",
            },
            Object {
              "cmd": "M 8",
              "image": Object {
                "process": [Function],
              },
              "stopAtEnd": true,
              "title": "Swing",
            },
            Object {
              "cmd": "M 14",
              "image": Object {
                "process": [Function],
              },
              "stopAtEnd": true,
              "title": "Tiptoe",
            },
            Object {
              "cmd": "M 20",
              "image": Object {
                "process": [Function],
              },
              "stopAtEnd": true,
              "title": "Ascend",
            },
            Object {
              "cmd": "M 5",
              "image": Object {
                "process": [Function],
              },
              "stopAtEnd": true,
              "title": "Up Down",
            },
            Object {
              "cmd": "M 13",
              "image": Object {
                "process": [Function],
              },
              "stopAtEnd": true,
              "title": "Flap Back",
            },
            Object {
              "cmd": "M 7",
              "image": Object {
                "process": [Function],
              },
              "stopAtEnd": true,
              "title": "Moonwalk Left",
            },
            Object {
              "cmd": "M 17",
              "image": Object {
                "process": [Function],
              },
              "stopAtEnd": true,
              "title": "Shake Left",
            },
            Object {
              "cmd": "M 15",
              "image": Object {
                "process": [Function],
              },
              "stopAtEnd": true,
              "title": "Bend Left",
            },
            Object {
              "cmd": "M 10",
              "image": Object {
                "process": [Function],
              },
              "stopAtEnd": true,
              "title": "Cross Left",
            },
            Object {
              "cmd": "M 12",
              "image": Object {
                "process": [Function],
              },
              "stopAtEnd": true,
              "title": "Flap Front",
            },
            Object {
              "cmd": "M 6",
              "image": Object {
                "process": [Function],
              },
              "stopAtEnd": true,
              "title": "Moonwalk Right",
            },
            Object {
              "cmd": "M 18",
              "image": Object {
                "process": [Function],
              },
              "stopAtEnd": true,
              "title": "Shake Right",
            },
            Object {
              "cmd": "M 16",
              "image": Object {
                "process": [Function],
              },
              "stopAtEnd": true,
              "title": "Bend Right",
            },
            Object {
              "cmd": "M 9",
              "image": Object {
                "process": [Function],
              },
              "stopAtEnd": true,
              "title": "Cross Right",
            },
          ],
        },
        Object {
          "category": "Sounds",
          "id": "sounds",
          "image": Object {
            "process": [Function],
          },
          "items": Array [
            Object {
              "cmd": "K 16",
              "image": Object {
                "process": [Function],
              },
              "stopAtEnd": false,
              "title": "Sing 1",
            },
            Object {
              "cmd": "K 17",
              "image": Object {
                "process": [Function],
              },
              "stopAtEnd": false,
              "title": "Sing 2",
            },
            Object {
              "cmd": "K 18",
              "image": Object {
                "process": [Function],
              },
              "stopAtEnd": false,
              "title": "Sing 3",
            },
            Object {
              "cmd": "K 4",
              "image": Object {
                "process": [Function],
              },
              "stopAtEnd": false,
              "title": "OhOoh",
            },
            Object {
              "cmd": "K 19",
              "image": Object {
                "process": [Function],
              },
              "stopAtEnd": false,
              "title": "Push",
            },
            Object {
              "cmd": "K 10",
              "image": Object {
                "process": [Function],
              },
              "stopAtEnd": false,
              "title": "Kinda Happy",
            },
            Object {
              "cmd": "K 8",
              "image": Object {
                "process": [Function],
              },
              "stopAtEnd": false,
              "title": "Happy",
            },
            Object {
              "cmd": "K 9",
              "image": Object {
                "process": [Function],
              },
              "stopAtEnd": false,
              "title": "Super Happy",
            },
            Object {
              "cmd": "K 11",
              "image": Object {
                "process": [Function],
              },
              "stopAtEnd": false,
              "title": "Sad",
            },
            Object {
              "cmd": "K 7",
              "image": Object {
                "process": [Function],
              },
              "stopAtEnd": false,
              "title": "Sleeping",
            },
            Object {
              "cmd": "K 3",
              "image": Object {
                "process": [Function],
              },
              "stopAtEnd": false,
              "title": "Surprise",
            },
            Object {
              "cmd": "K 2",
              "image": Object {
                "process": [Function],
              },
              "stopAtEnd": false,
              "title": "Disconnection",
            },
            Object {
              "cmd": "K 12",
              "image": Object {
                "process": [Function],
              },
              "stopAtEnd": false,
              "title": "Confused",
            },
            Object {
              "cmd": "K 15",
              "image": Object {
                "process": [Function],
              },
              "stopAtEnd": false,
              "title": "Fart",
            },
            Object {
              "cmd": "K 6",
              "image": Object {
                "process": [Function],
              },
              "stopAtEnd": false,
              "title": "Cuddly",
            },
          ],
        },
        Object {
          "category": "Gestures",
          "id": "gestures",
          "image": Object {
            "process": [Function],
          },
          "items": Array [
            Object {
              "cmd": "H 1",
              "image": Object {
                "process": [Function],
              },
              "stopAtEnd": false,
              "title": "Happy",
            },
            Object {
              "cmd": "H 2",
              "image": Object {
                "process": [Function],
              },
              "stopAtEnd": false,
              "title": "Super Happy",
            },
            Object {
              "cmd": "H 7",
              "image": Object {
                "process": [Function],
              },
              "stopAtEnd": false,
              "title": "Love",
            },
            Object {
              "cmd": "H 3",
              "image": Object {
                "process": [Function],
              },
              "stopAtEnd": false,
              "title": "Sad",
            },
            Object {
              "cmd": "H 13",
              "image": Object {
                "process": [Function],
              },
              "stopAtEnd": false,
              "title": "Fail",
            },
            Object {
              "cmd": "H 6",
              "image": Object {
                "process": [Function],
              },
              "stopAtEnd": false,
              "title": "Confused",
            },
            Object {
              "cmd": "H 5",
              "image": Object {
                "process": [Function],
              },
              "stopAtEnd": false,
              "title": "Fart",
            },
            Object {
              "cmd": "H 9",
              "image": Object {
                "process": [Function],
              },
              "stopAtEnd": false,
              "title": "Fretful",
            },
            Object {
              "cmd": "H 8",
              "image": Object {
                "process": [Function],
              },
              "stopAtEnd": false,
              "title": "Angry",
            },
            Object {
              "cmd": "H 4",
              "image": Object {
                "process": [Function],
              },
              "stopAtEnd": false,
              "title": "Sleeping",
            },
            Object {
              "cmd": "H 10",
              "image": Object {
                "process": [Function],
              },
              "stopAtEnd": false,
              "title": "Magic",
            },
            Object {
              "cmd": "H 12",
              "image": Object {
                "process": [Function],
              },
              "stopAtEnd": false,
              "title": "Victory",
            },
            Object {
              "cmd": "H 11",
              "image": Object {
                "process": [Function],
              },
              "stopAtEnd": false,
              "title": "Wave",
            },
          ],
        },
      ]
    }
  />
</Container>
`;
